simplified-app-shared/src/main/java/org/nypl/simplified/app/ReaderActivitySyncManager.kt

```diff
@@ -0,0 +1,173 @@
+package org.nypl.simplified.app
+
+import android.app.AlertDialog
+import android.content.Context
+import com.io7m.jfunctional.Some
+import org.nypl.simplified.app.reader.ReaderBookLocation
+import org.nypl.simplified.app.utilities.UIThread
+import org.nypl.simplified.books.core.AccountCredentials
+import org.nypl.simplified.books.core.AccountsControllerType
+import org.nypl.simplified.multilibrary.Account
+import org.nypl.simplified.opds.core.OPDSAcquisitionFeedEntry
+import org.readium.sdk.android.Package
+import org.slf4j.LoggerFactory
+import java.net.URI
+import java.util.Timer
+import kotlin.concurrent.schedule
+
+
+interface ReaderSyncManagerDelegate {
+  fun navigateToLocation(location: ReaderBookLocation)
```

Comment:
this looks useful - unrelated but just curious, how might one translate a ReaderPaginationChangedEvent  -> ReaderBookLocation? Now that this function exists (and does what I assume), you can probably save and restore`ReaderPaginationChangedEvents` on `onReadiumFunctionSettingsApplied`

Reply:
`ReaderPaginationChangedEvents` are fired off by the rendering engine each time the page is turned, and `navigateToLocation` will be utilizing a subset of that code to create an "openPage" event to take the user to another page in the book, based on the data that is syncronized with the server. Conceptually I'm kind of trying to keep these things separate in my head.

To address your question, "I don't know".. If you could let me know what you're thinking for why we'd want to change change the relationship of these rendering method/listeners you've listed.

Reply:
ah, so right now, when you change the text size or go to dusk mode/dark mode, it resets your location (you can confirm this by swiping a few pages in and changing the text size). It would good to save your location and restore it.